Installation Error: Mac OS X (El Capitan v.10.11.6)

Reporter: @craigmcdonnell

> tried installing netdata on Mac OS X (El Capitan v.10.11.6) today and it failed when attempting to chown on ~/opt/netdata/usr/libexec/netdata/plugins.d/apps.plugin. Creating an empty file at the specified path and then re-running the installer allowed installation to complete.

People verifying this bug:
- sebkim

Me, neither @ktsaou have access to the reported OS X version. If anyone is willing to make a PR, please do so.

Of course, here it is : 
```/Applications/Xcode.app/Contents/Developer/usr/bin/make  install-data-hook
if [ `id -u` == 0 ]; then \
		chown root '/opt/netdata/usr/libexec/netdata/plugins.d/apps.plugin' && \
		chmod 0755 '/opt/netdata/usr/libexec/netdata/plugins.d/apps.plugin' && \
		( setcap cap_dac_read_search,cap_sys_ptrace+ep '/opt/netdata/usr/libexec/netdata/plugins.d/apps.plugin' || \
		  chmod 4755 '/opt/netdata/usr/libexec/netdata/plugins.d/apps.plugin' ); \
	else \
		echo; \
		echo "ATTENTION"; \
		echo; \
		echo "/opt/netdata/usr/libexec/netdata/plugins.d/apps.plugin requires escalated capabilities:"; \
		echo "sudo chown root '/opt/netdata/usr/libexec/netdata/plugins.d/apps.plugin'"; \
		echo "sudo chmod 0755 '/opt/netdata/usr/libexec/netdata/plugins.d/apps.plugin'"; \
		echo "sudo setcap cap_dac_read_search,cap_sys_ptrace+ep '/opt/netdata/usr/libexec/netdata/plugins.d/apps.plugin'"; \
		echo; \
	fi
chown: /opt/netdata/usr/libexec/netdata/plugins.d/apps.plugin: No such file or directory
make[3]: *** [install-data-hook] Error 1
make[2]: *** [install-data-am] Error 2
make[1]: *** [install-am] Error 2
make: *** [install-recursive] Error 1
 FAILED
```

Looks a lot like issue #26 closed by #27

I see no mention of the installation of the apps.plugin. Here is the whole output of the `make install` : 

``` --- Install netdata ---
[netdata]# make install
Making install in charts.d
make[2]: Nothing to be done for `install-exec-am'.
 /usr/local/bin/gmkdir -p '/opt/netdata/usr/libexec/netdata/charts.d'
 /usr/local/bin/ginstall -c -m 644 README.md '/opt/netdata/usr/libexec/netdata/charts.d'
 /usr/local/bin/gmkdir -p '/opt/netdata/usr/libexec/netdata/charts.d'
 /usr/local/bin/ginstall -c ap.chart.sh apcupsd.chart.sh apache.chart.sh cpu_apps.chart.sh cpufreq.chart.sh example.chart.sh exim.chart.sh hddtemp.chart.sh load_average.chart.sh mem_apps.chart.sh mysql.chart.sh nginx.chart.sh nut.chart.sh opensips.chart.sh phpfpm.chart.sh postfix.chart.sh sensors.chart.sh squid.chart.sh tomcat.chart.sh '/opt/netdata/usr/libexec/netdata/charts.d'
Making install in conf.d
make[2]: Nothing to be done for `install-exec-am'.
 /usr/local/bin/gmkdir -p '/opt/netdata/etc/netdata/charts.d'
 /usr/local/bin/ginstall -c -m 644 charts.d/apache.conf charts.d/apcupsd.conf charts.d/cpufreq.conf charts.d/exim.conf charts.d/load_average.conf charts.d/mysql.conf charts.d/nut.conf charts.d/phpfpm.conf charts.d/sensors.conf charts.d/tomcat.conf charts.d/ap.conf charts.d/cpu_apps.conf charts.d/example.conf charts.d/hddtemp.conf charts.d/mem_apps.conf charts.d/nginx.conf charts.d/opensips.conf charts.d/postfix.conf charts.d/squid.conf '/opt/netdata/etc/netdata/charts.d'
 /usr/local/bin/gmkdir -p '/opt/netdata/etc/netdata'
 /usr/local/bin/ginstall -c -m 644 apps_groups.conf charts.d.conf fping.conf node.d.conf python.d.conf health_alarm_notify.conf health_email_recipients.conf stream.conf '/opt/netdata/etc/netdata'
 /usr/local/bin/gmkdir -p '/opt/netdata/etc/netdata/health.d'
 /usr/local/bin/ginstall -c -m 644 health.d/apache.conf health.d/backend.conf health.d/bind_rndc.conf health.d/elasticsearch.conf health.d/fping.conf health.d/haproxy.conf health.d/ipfs.conf health.d/isc_dhcpd.conf health.d/mdstat.conf health.d/memcached.conf health.d/mysql.conf health.d/named.conf health.d/nginx.conf health.d/postgres.conf health.d/redis.conf health.d/retroshare.conf health.d/squid.conf health.d/varnish.conf health.d/web_log.conf '/opt/netdata/etc/netdata/health.d'
 /usr/local/bin/gmkdir -p '/opt/netdata/etc/netdata/node.d'
 /usr/local/bin/ginstall -c -m 644 node.d/README.md node.d/named.conf.md node.d/sma_webbox.conf.md node.d/snmp.conf.md '/opt/netdata/etc/netdata/node.d'
 /usr/local/bin/gmkdir -p '/opt/netdata/etc/netdata/python.d'
 /usr/local/bin/ginstall -c -m 644 python.d/apache.conf python.d/apache_cache.conf python.d/bind_rndc.conf python.d/cpufreq.conf python.d/dovecot.conf python.d/elasticsearch.conf python.d/example.conf python.d/exim.conf python.d/fail2ban.conf python.d/freeradius.conf python.d/haproxy.conf python.d/hddtemp.conf python.d/ipfs.conf python.d/isc_dhcpd.conf python.d/mdstat.conf python.d/memcached.conf python.d/mysql.conf python.d/nginx.conf python.d/ovpn_status_log.conf python.d/phpfpm.conf python.d/postfix.conf python.d/postgres.conf python.d/redis.conf python.d/retroshare.conf python.d/sensors.conf python.d/squid.conf python.d/smartd_log.conf python.d/tomcat.conf python.d/varnish.conf python.d/web_log.conf '/opt/netdata/etc/netdata/python.d'
Making install in node.d
make[2]: Nothing to be done for `install-exec-am'.
 /usr/local/bin/gmkdir -p '/opt/netdata/usr/libexec/netdata/node.d'
 /usr/local/bin/ginstall -c -m 644 README.md named.node.js sma_webbox.node.js snmp.node.js '/opt/netdata/usr/libexec/netdata/node.d'
 /usr/local/bin/gmkdir -p '/opt/netdata/usr/libexec/netdata/node.d/node_modules'
 /usr/local/bin/ginstall -c -m 644 node_modules/netdata.js node_modules/extend.js node_modules/pixl-xml.js node_modules/net-snmp.js node_modules/asn1.js '/opt/netdata/usr/libexec/netdata/node.d/node_modules'
 /usr/local/bin/gmkdir -p '/opt/netdata/usr/libexec/netdata/node.d/node_modules/ber'
 /usr/local/bin/ginstall -c -m 644 node_modules/ber/index.js node_modules/ber/errors.js node_modules/ber/reader.js node_modules/ber/types.js node_modules/ber/writer.js '/opt/netdata/usr/libexec/netdata/node.d/node_modules/ber'
Making install in plugins.d
make[2]: Nothing to be done for `install-exec-am'.
 /usr/local/bin/gmkdir -p '/opt/netdata/usr/libexec/netdata/plugins.d'
 /usr/local/bin/ginstall -c -m 644 README.md '/opt/netdata/usr/libexec/netdata/plugins.d'
 /usr/local/bin/gmkdir -p '/opt/netdata/usr/libexec/netdata/plugins.d'
 /usr/local/bin/ginstall -c alarm-email.sh alarm-notify.sh alarm-test.sh cgroup-name.sh charts.d.dryrun-helper.sh charts.d.plugin fping.plugin node.d.plugin python.d.plugin tc-qos-helper.sh loopsleepms.sh.inc '/opt/netdata/usr/libexec/netdata/plugins.d'
Making install in python.d
make[2]: Nothing to be done for `install-exec-am'.
 /usr/local/bin/gmkdir -p '/opt/netdata/usr/libexec/netdata/python.d'
 /usr/local/bin/ginstall -c -m 644 README.md '/opt/netdata/usr/libexec/netdata/python.d'
 /usr/local/bin/gmkdir -p '/opt/netdata/usr/libexec/netdata/python.d'
 /usr/local/bin/ginstall -c apache.chart.py apache_cache.chart.py bind_rndc.chart.py cpufreq.chart.py cpuidle.chart.py dovecot.chart.py elasticsearch.chart.py example.chart.py exim.chart.py fail2ban.chart.py freeradius.chart.py haproxy.chart.py hddtemp.chart.py ipfs.chart.py isc_dhcpd.chart.py mdstat.chart.py memcached.chart.py mysql.chart.py nginx.chart.py ovpn_status_log.chart.py phpfpm.chart.py postfix.chart.py postgres.chart.py redis.chart.py retroshare.chart.py sensors.chart.py squid.chart.py smartd_log.chart.py tomcat.chart.py varnish.chart.py web_log.chart.py python-modules-installer.sh '/opt/netdata/usr/libexec/netdata/python.d'
 /usr/local/bin/gmkdir -p '/opt/netdata/usr/libexec/netdata/python.d/python_modules'
 /usr/local/bin/ginstall -c -m 644 python_modules/__init__.py python_modules/base.py python_modules/msg.py python_modules/lm_sensors.py '/opt/netdata/usr/libexec/netdata/python.d/python_modules'
 /usr/local/bin/gmkdir -p '/opt/netdata/usr/libexec/netdata/python.d/python_modules/pyyaml2'
 /usr/local/bin/ginstall -c -m 644 python_modules/pyyaml2/__init__.py python_modules/pyyaml2/composer.py python_modules/pyyaml2/constructor.py python_modules/pyyaml2/cyaml.py python_modules/pyyaml2/dumper.py python_modules/pyyaml2/emitter.py python_modules/pyyaml2/error.py python_modules/pyyaml2/events.py python_modules/pyyaml2/loader.py python_modules/pyyaml2/nodes.py python_modules/pyyaml2/parser.py python_modules/pyyaml2/reader.py python_modules/pyyaml2/representer.py python_modules/pyyaml2/resolver.py python_modules/pyyaml2/scanner.py python_modules/pyyaml2/serializer.py python_modules/pyyaml2/tokens.py '/opt/netdata/usr/libexec/netdata/python.d/python_modules/pyyaml2'
 /usr/local/bin/gmkdir -p '/opt/netdata/usr/libexec/netdata/python.d/python_modules/pyyaml3'
 /usr/local/bin/ginstall -c -m 644 python_modules/pyyaml3/__init__.py python_modules/pyyaml3/composer.py python_modules/pyyaml3/constructor.py python_modules/pyyaml3/cyaml.py python_modules/pyyaml3/dumper.py python_modules/pyyaml3/emitter.py python_modules/pyyaml3/error.py python_modules/pyyaml3/events.py python_modules/pyyaml3/loader.py python_modules/pyyaml3/nodes.py python_modules/pyyaml3/parser.py python_modules/pyyaml3/reader.py python_modules/pyyaml3/representer.py python_modules/pyyaml3/resolver.py python_modules/pyyaml3/scanner.py python_modules/pyyaml3/serializer.py python_modules/pyyaml3/tokens.py '/opt/netdata/usr/libexec/netdata/python.d/python_modules/pyyaml3'
Making install in src
 /usr/local/bin/gmkdir -p '/opt/netdata/usr/sbin'
  /usr/local/bin/ginstall -c netdata '/opt/netdata/usr/sbin'
 /usr/local/bin/gmkdir -p '/opt/netdata/var/cache/netdata'
 /usr/local/bin/ginstall -c -m 644 .keep '/opt/netdata/var/cache/netdata'
 /usr/local/bin/gmkdir -p '/opt/netdata/var/log/netdata'
 /usr/local/bin/ginstall -c -m 644 .keep '/opt/netdata/var/log/netdata'
 /usr/local/bin/gmkdir -p '/opt/netdata/var/lib/netdata/registry'
 /usr/local/bin/ginstall -c -m 644 .keep '/opt/netdata/var/lib/netdata/registry'
 /usr/local/bin/gmkdir -p '/opt/netdata/var/lib/netdata'
 /usr/local/bin/ginstall -c -m 644 .keep '/opt/netdata/var/lib/netdata'
/Applications/Xcode.app/Contents/Developer/usr/bin/make  install-data-hook
if [ `id -u` == 0 ]; then \
		chown root '/opt/netdata/usr/libexec/netdata/plugins.d/apps.plugin' && \
		chmod 0755 '/opt/netdata/usr/libexec/netdata/plugins.d/apps.plugin' && \
		( setcap cap_dac_read_search,cap_sys_ptrace+ep '/opt/netdata/usr/libexec/netdata/plugins.d/apps.plugin' || \
		  chmod 4755 '/opt/netdata/usr/libexec/netdata/plugins.d/apps.plugin' ); \
	else \
		echo; \
		echo "ATTENTION"; \
		echo; \
		echo "/opt/netdata/usr/libexec/netdata/plugins.d/apps.plugin requires escalated capabilities:"; \
		echo "sudo chown root '/opt/netdata/usr/libexec/netdata/plugins.d/apps.plugin'"; \
		echo "sudo chmod 0755 '/opt/netdata/usr/libexec/netdata/plugins.d/apps.plugin'"; \
		echo "sudo setcap cap_dac_read_search,cap_sys_ptrace+ep '/opt/netdata/usr/libexec/netdata/plugins.d/apps.plugin'"; \
		echo; \
	fi
chown: /opt/netdata/usr/libexec/netdata/plugins.d/apps.plugin: No such file or directory
make[3]: *** [install-data-hook] Error 1
make[2]: *** [install-data-am] Error 2
make[1]: *** [install-am] Error 2
make: *** [install-recursive] Error 1
 FAILED
```

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -158,19 +158,3 @@ apps_plugin_LDADD = \
 	$(OPTIONAL_CAP_LIBS) \
 	$(NULL)
 
-install-data-hook:
-	if [ `id -u` == 0 ]; then \
-		chown root '$(DESTDIR)$(pluginsdir)/apps.plugin' && \
-		chmod 0755 '$(DESTDIR)$(pluginsdir)/apps.plugin' && \
-		( setcap cap_dac_read_search,cap_sys_ptrace+ep '$(DESTDIR)$(pluginsdir)/apps.plugin' || \
-		  chmod 4755 '$(DESTDIR)$(pluginsdir)/apps.plugin' ); \
-	else \
-		echo; \
-		echo "ATTENTION"; \
-		echo; \
-		echo "$(pluginsdir)/apps.plugin requires escalated capabilities:"; \
-		echo "sudo chown root '$(DESTDIR)$(pluginsdir)/apps.plugin'"; \
-		echo "sudo chmod 0755 '$(DESTDIR)$(pluginsdir)/apps.plugin'"; \
-		echo "sudo setcap cap_dac_read_search,cap_sys_ptrace+ep '$(DESTDIR)$(pluginsdir)/apps.plugin'"; \
-		echo; \
-	fi

--- a/src/freebsd_sysctl.c
+++ b/src/freebsd_sysctl.c
@@ -382,18 +382,18 @@ int do_freebsd_sysctl(int update_every, usec_t dt) {
                 if (unlikely(!st)) {
                     st = rrdset_create_localhost("system", "cpu", NULL, "cpu", "system.cpu", "Total CPU utilization", "percentage", 100, update_every, RRDSET_TYPE_STACKED);
 
-                    rrddim_add(st, "user", NULL, 1, 1, RRD_ALGORITHM_PCENT_OVER_DIFF_TOTAL);
                     rrddim_add(st, "nice", NULL, 1, 1, RRD_ALGORITHM_PCENT_OVER_DIFF_TOTAL);
                     rrddim_add(st, "system", NULL, 1, 1, RRD_ALGORITHM_PCENT_OVER_DIFF_TOTAL);
+                    rrddim_add(st, "user", NULL, 1, 1, RRD_ALGORITHM_PCENT_OVER_DIFF_TOTAL);
                     rrddim_add(st, "interrupt", NULL, 1, 1, RRD_ALGORITHM_PCENT_OVER_DIFF_TOTAL);
                     rrddim_add(st, "idle", NULL, 1, 1, RRD_ALGORITHM_PCENT_OVER_DIFF_TOTAL);
                     rrddim_hide(st, "idle");
                 }
                 else rrdset_next(st);
 
-                rrddim_set(st, "user", cp_time[0]);
                 rrddim_set(st, "nice", cp_time[1]);
                 rrddim_set(st, "system", cp_time[2]);
+                rrddim_set(st, "user", cp_time[0]);
                 rrddim_set(st, "interrupt", cp_time[3]);
                 rrddim_set(st, "idle", cp_time[4]);
                 rrdset_done(st);
@@ -425,18 +425,18 @@ int do_freebsd_sysctl(int update_every, usec_t dt) {
                             st = rrdset_create_localhost("cpu", cpuid, NULL, "utilization", "cpu.cpu", "Core utilization",
                                                "percentage", 1000, update_every, RRDSET_TYPE_STACKED);
 
-                            rrddim_add(st, "user", NULL, 1, 1, RRD_ALGORITHM_PCENT_OVER_DIFF_TOTAL);
                             rrddim_add(st, "nice", NULL, 1, 1, RRD_ALGORITHM_PCENT_OVER_DIFF_TOTAL);
                             rrddim_add(st, "system", NULL, 1, 1, RRD_ALGORITHM_PCENT_OVER_DIFF_TOTAL);
+                            rrddim_add(st, "user", NULL, 1, 1, RRD_ALGORITHM_PCENT_OVER_DIFF_TOTAL);
                             rrddim_add(st, "interrupt", NULL, 1, 1, RRD_ALGORITHM_PCENT_OVER_DIFF_TOTAL);
                             rrddim_add(st, "idle", NULL, 1, 1, RRD_ALGORITHM_PCENT_OVER_DIFF_TOTAL);
                             rrddim_hide(st, "idle");
                         } else
                             rrdset_next(st);
 
-                        rrddim_set(st, "user", pcpu_cp_time[i * 5 + 0]);
                         rrddim_set(st, "nice", pcpu_cp_time[i * 5 + 1]);
                         rrddim_set(st, "system", pcpu_cp_time[i * 5 + 2]);
+                        rrddim_set(st, "user", pcpu_cp_time[i * 5 + 0]);
                         rrddim_set(st, "interrupt", pcpu_cp_time[i * 5 + 3]);
                         rrddim_set(st, "idle", pcpu_cp_time[i * 5 + 4]);
                         rrdset_done(st);

